rfd/9999-webauthn-support.md

```diff
@@ -0,0 +1,674 @@
+               │ *taps*       │                                  │                                           │   
+           <───┘              │                                  │                                           │   
+           │                  │                                  │                                           │   
+           │                  │     POST /webapi/u2f/certs       │                                           │   
+           │                  │─────────────────────────────────>│                                           │   
+           │                  │                                  │                                           │   
+           │                  │                                  │POST /v2/users/codingllama/ssh/authenticate│   
+           │                  │                                  │───────────────────────────────────────────>   
+           │                  │                                  │                                           │   
+           │                  │                                  │         lib.auth.SSHLoginResponse         │   
+           │                  │                                  │<───────────────────────────────────────────   
+           │                  │                                  │                                           │   
+           │                  │    lib.auth.SSHLoginResponse     │                                           │   
+           │                  │<─────────────────────────────────│                                           │   
+           │                  │                                  │                                           │   
+           │      done        │                                  │                                           │   
+           │<─────────────────│                                  │                                           │   
+      codingllama           ┌─┴─┐                             ┌──┴──┐                                      ┌─┴──┐
+          ┌─┐               │tsh│                             │proxy│                                      │auth│
+          ║"│               └───┘                             └─────┘                                      └────┘
+          └┬┘                                                                                                    
+          ┌┼┐                                                                                                    
+           │                                                                                                     
+          ┌┴┐                                                                                                    
+```
+
+The proposed changes are twofold:
+
+1. Aliasing the current U2F-centric endpoints to more generic MFA-named
+   counterparts
+
+2. Modifying existing messages to support WebAuthn as a login method
+
+Endpoint changes are:
```

Comment:
why not switch this to GRPC + tls SNI? this will let the client talk to the auth server and not the proxy

Reply:
I thought you might suggest this. :)

I agree that switching to gRPC is desirable, but I am concerned about changing too much at once. I'd rather push webauthn support first, then refactor the login in a separate change. Moving it all to gRPC is a larger refactor that will touch a good chunk of the local login flow, whereas webauthn support is a more focused change with little impact to existing endpoints.

Reply:
Sounds good to me, let's just create a ticket, so we don't forget.

Reply:
Done - https://github.com/gravitational/teleport/issues/7966.